starfish/core/intensity_table/intensity_table_coordinates.py

```diff
@@ -1,46 +1,63 @@
+from typing import Mapping, Optional
+
 import numpy as np
 import xarray as xr
 
 from starfish.core.imagestack.imagestack import ImageStack
 from starfish.core.intensity_table.intensity_table import IntensityTable
-from starfish.core.types import Axes, Coordinates, Features
+from starfish.core.types import Axes, Coordinates, Features, SpotFindingResults
 
 
-def transfer_physical_coords_from_imagestack_to_intensity_table(
-        image_stack: ImageStack, intensity_table: IntensityTable
-) -> IntensityTable:
+def transfer_physical_coords_to_intensity_table(*, intensity_table: IntensityTable,
+                                                image_stack: Optional[ImageStack] = None,
+                                                spots: Optional[SpotFindingResults] = None,
+                                                ) -> IntensityTable:
     """
-    Transfers physical coordinates from an Imagestack's coordinates xarray to an intensity table
+    Transfers physical coordinates from either an Imagestack or SpotFindingResults to an
+    intensity table
 
     1. Creates three new coords on the intensity table (xc, yc, zc)
     2. For every spot:
         - Get pixel x,y values
         - Calculate the physical x,y values
         - Assign those values to the coords arrays for this spot
     """
-    # TODO shanaxel42 consider refactoring pixel case where were can just reshape from Imagestack
 
     pairs = (
         (Axes.X.value, Coordinates.X.value),
         (Axes.Y.value, Coordinates.Y.value),
         (Axes.ZPLANE.value, Coordinates.Z.value)
     )
 
+    if image_stack is None and spots is None:
+        raise ValueError("Must provide either SpotFindingResults or ImageStack to "
+                         "calculate coordinates from")
+
+    coord_ranges: Mapping[str, xr.DataArray]
```

Comment:
As I discovered last night, the newest version of xarray will probably want this as:

```suggestion
    coord_ranges: Mapping[Hashable, xr.DataArray]
```